### ISSUE TITLE
Consider using tooltips to help make specific terms more accessible 

### ISSUE THREAD

**Issue by bivens-dev** [dart-lang/site-www#4832]
### Describe the problem

![IMG_8406](https://github.com/dart-lang/site-www/assets/113126707/1ed1a46b-ae46-45d1-932f-ba481ca836e8)

I’ve included the following example here as one that while is technically great it contains a lot of terms with very specific meanings that may not be familiar to a meaningful portion of the target audience.

One pattern we could employ both here and across the website is the use of tooltips that help provide the reader with a richer context and therefore a deeper understanding.

**Comment by parlough** [comment]
We are [considering](https://github.com/dart-lang/site-www/issues/4622) adding an extended Glossary of Dart-specific which would possibly support tooltips. 

However that is focused on Dart-specific terms, but it seems in this case you mean some non-Dart words. Are there some words in particular that you think would benefit from this here? Perhaps they should be replaced with easier to understand terms or definitions, since adding tooltips for non-Dart specific words might get a bit overwhelming and tooltips are often less accessible. 

I do think it would be beneficial to avoid or at least augment terms like aggregate and heterogenous here.

**Comment by bivens-dev** [comment]
The glossary concept sounds like it makes a lot of sense and I see in there that you were already considering a similar design pattern as well.

Is it limited to Dart specific just purely as an exercise to make it a more manageable task or another reason? I saw some concerns highlighted about fears of it becoming a dumping ground which were understandable but it doesn’t feel like it would be too difficult to find reasonable use cases that weren’t purely dart specific too.

Another pattern that might make sense here although it’s a bit specific would be to create some additional content that is perhaps somewhere in the middle such as a mental model to help developers understand how to think about the decision process for selecting the most appropriate data type and what kinds of considerations and trade offs they should be aware of.

It would be an example of the kind of content that helps support these individual feature specific pages and help put them into a wider context while still keeping things very Dart specific (I.e here’s how to model your data in Dart).

Not sure if that feels particularly out of scope for what you folks had in mind though?

**Comment by parlough** [comment]
Yeah I guess the concern is where to decide what words to include and what words to not include. We don't want the site to become a dictionary or _need_ a dictionary, especially since a lot of browsers and operating systems include hover/select then "look up" functionality now. 

I generally think if text is not understandable (to native and non-native speakers) due to complicated word or phrasing, the text should probably be adjusted or restructured. I prefer avoiding the need to look up words, either in the a tooltip, glossary, or dictionary if possible. If it has a specific meaning in a Dart context or occurs a lot throughout the site, I can see value in adding it though.

As for your mention of other content to include, such as how to choose a specific data type, it sounds like a great idea to include some context there. In terms of records we could add more context or examples on why to choose them over other collection types. It would definitely make the content more approachable and understandable, for both those new and experienced in Dart.

We currently use this language documentation more for dives into specific features and for developers already familiar with programming. We are currently figuring out our exact future direction for this, but the goal is to introduce more "Learn Dart" material such as what you are describing, which will better cover integrating features and building those mental models. I really do love the idea of a guide exploring how to model your data in Dart!!

**Comment by antfitch** [comment]
Tentatively closing this due to the length of time that has passed and the changes that have since been made to the site.
